### PR TITLE
feat!: accept bundles instead of just certificates

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This PR adds the change required to accept and distribute issuer certificates alongside the certificate of the CSR. A new certificate must contain a bundle of at least 2 certificates in a valid chain to be accepted. Any extra characters will be ignored.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: internal/certdb/validation.go [hunk 1]
```diff
@@ -27,27 +30,58 @@ func ValidateCertificateRequest(csr string) error {
...
-		return errors.New("given PEM string not a certificate")
+	if len(certificates) < 2 {
+		return errors.New("less than 2 certificate PEM strings were found")
 	}
-	_, err := x509.ParseCertificate(certBlock.Bytes)
-	if err != nil {
-		return err
+	for i := 0; i < len(certificates)-1; i++ {
+		firstCert := certificates[i]
+		secondCert := certificates[i+1]
```
Comment (ghislainbourgeois) [inline internal/certdb/validation.go:64]:
```suggestion
	for i, firstCert := range certificates[:len(certificates)-1] {
		secondCert := certificates[i+1]
```

File: internal/certdb/validation.go [hunk 2]
```diff
@@ -27,27 +30,58 @@ func ValidateCertificateRequest(csr string) error {
...
+		if !bytes.Equal(firstCert.RawIssuer, secondCert.RawSubject) {
+			return fmt.Errorf("invalid certificate chain: certificate %d, certificate %d: subjects do not match", i, i+1)
+		}
+		if err := firstCert.CheckSignatureFrom(secondCert); err != nil {
+			return fmt.Errorf("invalid certificate chain: certificate %d, certificate %d: keys do not match: %s", i, i+1, err.Error())
+		}
+	}
+	for i := 1; i < len(certificates)-1; i++ {
+		cert := certificates[i]
```
Comment (ghislainbourgeois) [inline internal/certdb/validation.go:73]:
```suggestion
	for i, cert := range certificates[:len(certificates)-1] {
```

File: internal/api/handlers_test.go [hunk 3]
```diff
@@ -349,7 +413,7 @@ func TestGoCertCertificatesHandlers(t *testing.T) {
 		},
 	}
 	for _, tC := range testCases {
-		t.Run(tC.desc, func(t *testing.T) {
+		t.Run(fmt.Sprintf("step %d: %s", tC.step, tC.desc), func(t *testing.T) {
```
Comment (ghislainbourgeois) [inline internal/api/handlers_test.go:416]: I would remove the step field and simply take this from the range:

```suggestion
    for step, tC := range testCases {
		t.Run(fmt.Sprintf("step %d: %s", tC.step, tC.desc), func(t *testing.T) {
```

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: internal/api/handlers_test.go (hunk 3 above)
Comment (kayra1) [reply internal/api/handlers_test.go:416]: I liked having the steps because it helped scrolling to find the exact case that went wrong.

Comment (ghislainbourgeois) [reply internal/api/handlers_test.go:416]: Can't the same be done achieved with the description? This is making the file 26 lines longer at the moment, and will be a pain to maintain cleanly if we decide to remove a test case.

Comment (kayra1) [reply internal/api/handlers_test.go:416]: that's not a bad idea. changed.

File: internal/certdb/validation.go (hunk 2 above)
Comment (kayra1) [reply internal/certdb/validation.go]: this one actually starts at 1, so i used a range loop with the slice starting from 1.

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**